Assets/Scripts/Interaction/PaintableSurface.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections;
+using Interaction.Core;
+using Inventory.Custom;
+using Inventory.UI;
+using ItemComponents;
+using UnityEngine;
+
+namespace Interaction
+{
+    /// <summary>
+    /// Component for handling the Paint action targeted at an object.
+    /// </summary>
+    [RequireComponent(typeof(InteractionReceiver))]
+    public class PaintableSurface : MonoBehaviour, IContinuousInteraction
```

Comment:
If this doesn't use the monobehaviour at all, then it might as well be a Scriptable Object

Reply:
Not sure what you mean? It has a lot of logic contained within it, and it needs to be attached to any object that can be painted on.